core/src/View/Kokkos_BasicView.hpp

```diff
@@ -127,6 +127,13 @@ KOKKOS_INLINE_FUNCTION constexpr auto accessor_from_mapping_and_accessor_arg(
   return AccessorType(arg.value);
 }
 
+template <typename DataHandleType, typename MappingType, typename AccessorType>
+constexpr DataHandleType data_handle_from_allocation(
+    SharedAllocationRecord<void, void> *rec, const MappingType & /* mapping */,
+    const AccessorType & /* accessor */) {
+  return DataHandleType(rec);
+}
+
```

Comment:
are you saying you need the mapping to construct your data handle? Why can't you store the extra information you need in the accessor? Why does it go with the data handle?